src/legacy/ui/public/agg_types/buckets/_inline_comp_wrapper.js

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+
+const wrapWithInlineComp = Component => props => (
+  <div className={`visEditorAggParam--inline visEditorAggParam--inline-${props.aggParam.name}`}>
```

Comment:
I think that if you go the route of renaming this to explicitly state `half` then you won't need to depend on the `props.aggParam.name` and the margin can be added to a sibling CSS selector like: `.visEditorAggParam--half + .visEditorAggParam--half`

Reply:
Unfortunately the solution wouldn't work because of our current structure (both them are built through the existing interface and are wrapped into additional elements). They are not siblings and that was the main pain of positioning them both since the start.

![order-and-size-siblings](https://user-images.githubusercontent.com/31325372/56645419-4fae8900-6686-11e9-8a21-7db2af5ac634.PNG)

Reply:
@cchaos are you OK with current implementation ?

Reply:
I just have a concern around using the aggParam name.

Will `props.aggParam.name` eve have a space in it like "name here"? If so you'll end up with a className list like: `visEditorAggParam--inline visEditorAggParam--inline-name here` which causes `.here` to be a valid class which is not good.

I also still think you can change the class from `--inline` to `--half` so it's easily understood that the CSS will make that field 50% wide.

Reply:
> Will `props.aggParam.name` eve have a space in it like "name here"? If so you'll end up with a className list like: `visEditorAggParam--inline visEditorAggParam--inline-name here` which causes `.here` to be a valid class which is not good.

It never can have s space in a name because we are using these names in a url string. These names should always have or camelCase style or `_` between words, like `intervalBase`, `ip_range`, `min_doc_count`, `extended_bounds` and so on.

> I also still think you can change the class from `--inline` to `--half` so it's easily understood that the CSS will make that field 50% wide.

This work has been already done in this [commit](https://github.com/elastic/kibana/pull/35134/commits/cd300f6cbe8b36cd57553c5fe026e64389c17b23)